drivers/gpio/gpio_lmp90xxx.c

```diff
@@ -139,12 +138,10 @@ static int gpio_lmp90xxx_pin_interrupt_configure(const struct device *dev,
 static int gpio_lmp90xxx_init(const struct device *dev)
 {
 	const struct gpio_lmp90xxx_config *config = dev->config;
-	struct gpio_lmp90xxx_data *data = dev->data;
 
-	data->parent = device_get_binding(config->parent_dev_name);
-	if (!data->parent) {
-		LOG_ERR("parent LMP90xxx device '%s' not found",
-			config->parent_dev_name);
+	if (!device_is_ready(config->parent)) {
+		LOG_ERR("parent LMP90xxx device '%s' not ready",
+			log_strdup(config->parent->name));
```

Comment:
Is it actually needed to duplicate this string? It is supposed to be a permanent one, right?

Reply:
Thanks, force of habit when seeing a `%s` in a `LOG_*()` macro. Fixed.